File: server/src/main/java/by/brawl/ws/dto/HeroDto.java
package by.brawl.ws.dto;

import by.brawl.util.Mappers;
import by.brawl.ws.holder.HeroHolder;

import java.util.LinkedHashSet;
import java.util.Set;

public class HeroDto {
    private String id;
    private Set<BodypartDto> bodyparts = new LinkedHashSet<>();
    private Set<EquipmentDto> equipments = new LinkedHashSet<>();

    public HeroDto(HeroHolder heroHolder) {
        id = heroHolder.getId();
        bodyparts = Mappers.asSet(heroHolder.getBodyparts(), BodypartDto::new);
        equipments = Mappers.asSet(heroHolder.getEquipments(), EquipmentDto::new);
//        spells.addAll(
//                heroHolder.getAllSpells().stream()
//                        .map(SpellDto::new)
//                        .collect(Collectors.toSet())
//        );
    }

    public String getId() {
        return id;
    }

    public Set<BodypartDto> getBodyparts() {
        return bodyparts;
    }

    public Set<EquipmentDto> getEquipments() {
        return equipments;
    }
    //    public Set<SpellDto> getSpells() {
//        return spells;
//    }
}

File: server/src/main/java/by/brawl/ws/dto/HeroDto.kt
package by.brawl.ws.dto

import by.brawl.ws.holder.HeroHolder
import java.util.*

class HeroDto(heroHolder: HeroHolder) {

    val id: String = heroHolder.id
    val name: String = heroHolder.name
    var spells = heroHolder.allSpells.map { SpellDto(it) }.toMutableList()
    var bodyparts = heroHolder.bodyparts.map { BodypartDto(it) }.toCollection(LinkedHashSet<BodypartDto>())
    var equipments = heroHolder.equipments.map { EquipmentDto(it) }.toCollection(LinkedHashSet<EquipmentDto>())
    val attributes = heroHolder.attributes

}